### ISSUE TITLE
p-dropdown: associate a label with this component (with for="" property)

### ISSUE THREAD

**Issue by fperie** [primefaces/primeng#14733]
### Describe the bug

Hello,
I tried to set the inputId property of the Dropdown component to associate it with a label using the for attribute. This does not work because the Dropdown component does not have an input field. However, for accessibility purposes, it would be necessary to have a readonly input field. Do you plan to change this behavior?

### Environment

PrimeNG 17.5.0

### Angular version

17.1.2

### PrimeNG version

17.5.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

18.16.1

### Steps to reproduce the behavior

If i have a label of dropdown field, its not getting announced with screen reader. Once I choose any option, the label is getting override with selected value and reader reads the field with selected value.

```html
<label for="fieldId"> Field</label>
<p-dropdown [options]="cities" formControlName="
fieldName" aria-required="true" inputId="fieldId"></p-dropdown>
```

If I click on label, the dropdown field doesn't have the focus.

**Comment by danielzeliazkow** [comment]
[attr.id]="id" is now on div element instead of input in dropdown.ts

**Comment by cetincakiroglu** [comment]
Hi,

Yes we're planning to change this behavior, we'll add a read-only input in the future as accessibility enhancement.

**Comment by athulkrizz** [comment]
Hi, Any updates on this? Its been an year and issue is still there.

**Comment by nZeus** [comment]
`p-select` has `inputId` which does nothing since there is no `<input />` element...

**Comment by oksanakritskaya** [comment]
I think it’s more appropriate to focus the selected (but not opened) dropdown when the label is clicked - this is native HTML5 behavior.
I fixed it like this:
```
<label for="fieldId" (click)="fieldSelect.focus()">Field</label>
<p-dropdown
    #fieldSelect
    [options]="cities"
    formControlName="fieldName"
    aria-required="true"
    inputId="fieldId"
></p-dropdown>
```

**Comment by Iancovski** [comment]
> Hi,
> 
> Yes we're planning to change this behavior, we'll add a read-only input in the future as accessibility enhancement.

@cetincakiroglu Hey, it's been 2 years, the `p-dropdown` was replaced by `p-select`, but the situation persists. Is there any timeline for a solution?
